pages/01_metadata.py

```diff
@@ -0,0 +1,156 @@
+# Define the page's content within a variable called layout or
+# a function called layout that returns the content.
+#
+# File upload component:
+# https://dash.plotly.com/dash-core-components/upload
+
+import base64
+import os
+
+import dash
+import dash_bootstrap_components as dbc
+import pandas as pd
+import yaml
+from dash import Input, Output, State, callback, dcc, html
+
+###############
+# Register this page
+dash.register_page(__name__)
+
+
+####################
+# Read yamls in dir as pandas dataframe
+
+
+def df_from_metadata_yaml_files(parent_dir):
+    # TODO: refactor, I think it could be more compact (see below)
+
+    list_metadata_files = [
+        el for el in os.listdir(parent_dir) if el.endswith("metadata.yaml")
+    ]
+
+    list_df_metadata = []
+    for yl in list_metadata_files:
+        with open(os.path.join(parent_dir, yl)) as f:
+            list_df_metadata.append(
+                pd.DataFrame.from_dict(yaml.safe_load(f), orient="index")
+            )
+    return pd.concat(
+        list_df_metadata, ignore_index=True, axis=1
+    ).T  # TODO review why transpose required
+
+
+# TODO: previous approach with json, can I do it as compact?
+# read_fn = lambda x: pd.read_json(os.path.join(parent_dir, x), orient="index")
+# df = pd.concat(map(read_fn, list_metadata_files), ignore_index=True, axis=1)
+
+
+###############
+# Define table component from pandas dataframe
+# TODO: check table style options (see Table class)
+
+
+def generate_tbl_component(parent_dir):
+    # ##########################
+    # Using html dash components
+    # dataframe = df_from_metadata_yaml_files(parent_dir)
+    # table = html.Table(
+    #     children =
+    #     [ # define table head?
+    #        html.Thead(html.Tr([html.Th(col, style={'width':'15%'})
+    #                               for col in dataframe.columns])),
+    #        # define table body
+    #        html.Tbody(
+    #            [
+    #               html.Tr(
+    #                 [
+    #                   html.Td(dataframe.iloc[i][col], style={'width':'15%'})
+    #                   for col in dataframe.columns
+    #                  ]
+    #                 )
+    #                 #range(min(len(dataframe), max_rows))
+    #                 for i in range(len(dataframe))
+    #             ]
+    #         ),
+    #     ],
+    #     style={'width':'100%',}
+    # )
+    #########
+
+    #########
+    # Using bootstrap components
+    table = dbc.Table.from_dataframe(
```

Comment:
Some of the description fields in the table contain very long strings, resulting in very thick rows. Any chance we could remedy that by only displaye e.g. the first n characters of a string and only show the full string on hover/preview (if dash includes that functionality)